ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces/ethernet_interfaces.py

```diff
@@ -25,21 +27,30 @@ def ethernet_interfaces(self) -> dict | None:
 
                 # Remove None values
                 ethernet_interface = {key: value for key, value in ethernet_interface.items() if value is not None}
-
-                interface_name = p2p_link["data"]["interface"]
-                ethernet_interfaces[interface_name] = ethernet_interface
+                if p2p_link["data"]["interface"] in interface_names:
```

Comment:
So as I read this, you are replacing duplicate interfaces to keep the old behavior. We should not. If ppl have duplicate interfaces, we should trigger an error like we have implemented under network services for vrfs, vlans and vnis.
The same should be done below (use the name list of interface names) and also in l3_edge.